modules/infra/control/iface.c

```diff
@@ -34,10 +34,13 @@ void iface_type_register(struct iface_type *type) {
 	STAILQ_INSERT_TAIL(&types, type, next);
 }
 
+#define IFACE_ID_FIRST GR_IFACE_ID_UNDEF + 1
+
+// the first slot is wasted by GR_IFACE_ID_UNDEF
 static struct iface **ifaces;
 
 static int next_ifid(uint16_t *ifid) {
-	for (uint16_t i = 0; i < MAX_IFACES; i++) {
+	for (uint16_t i = IFACE_ID_FIRST; i < MAX_IFACES; i++) {
```

Comment:
Should we set GR_IFACE_ID_UNDEF to -1, and have MAX_IFACE to ((1<<16) -1) ?

Reply:
My intent was to ensure that non-initialized values of iface_id (0) are invalid.